PT-1435 Platform services should be able to start without MongoDb

CedricPerotto opened the pull request:
Database first connection is now launched through a goroutine that doesn't block the service startup. Database session is available once this connection has been established. 
EnsureIndexes methods are also done in the goroutine (that may be the only reason we need connection on startup).

Two new environment variables are introduced :
- TIDEPOOL_STORE_WAIT_CONNECTION_INTERVAL (default 5) wait interval in seconds for the goroutine between failure on the last attempt and next attempt
 - TIDEPOOL_STORE_MAX_CONNECTION_ATTEMPTS (default 0) number of attempts before the goroutine crashes (and so on the service). If this number is <= 0, it will loop infinitely until it is able to connect

CedricPerotto committed:
Mongo store : connection in goroutine

CedricPerotto committed:
platform-data implementation

CedricPerotto committed:
platform-data updated tests

CedricPerotto committed:
platform-auth implementation +tests

CedricPerotto committed:
platform-blob implementation + test

CedricPerotto committed:
platform-image implementation + test

CedricPerotto committed:
platform-task implementation + test

CedricPerotto committed:
platform-user implementation + test

CedricPerotto committed:
Confirmation store update (EnsureIndexes) + test

CedricPerotto committed:
Test updates on remaining stores

CedricPerotto committed:
Make log/test/serializer concurrent safe

CedricPerotto committed:
Fixing profile/store/structured/mongo/

CedricPerotto committed:
Changelog

franck-fourel reviewed (commented):
couple of general comments:
- we were testing the definition of the indexes in previous version which is not done anymore as . Not really sure it's necessary as function to create indexes is tested.
- No error in the logs when the routine cannot join the database. Would be great to have at least the retry attempts in the log and the status when closingChannel is set to true. 
- wondering as well if we should add test on the goroutine

CedricPerotto committed:
Making goroutine testable & Add a test. Add logs on connection failure/success

CedricPerotto commented:
> couple of general comments:
> 
> * we were testing the definition of the indexes in previous version which is not done anymore as . Not really sure it's necessary as function to create indexes is tested.

I didn't think it was necessary and didn't add any tests... 

> * No error in the logs when the routine cannot join the database. Would be great to have at least the retry attempts in the log and the status when closingChannel is set to true.

I added logs on connection success/failure 

> * wondering as well if we should add test on the goroutine

I added a test (and had to refactor the code a little to make it testable).

CedricPerotto committed:
Secure Config.Addresses concurrent accesses (Data Race Fix)

CedricPerotto committed:
Fixing permission & tools store Config.Addresses calls

franck-fourel reviewed (approved):
LGTM

franck-fourel commented:
@CedricPerotto as I guess we want to release it, can you please update the changelog so that it can be realeased ?

CedricPerotto committed:
Bump version

CedricPerotto commented:
> @CedricPerotto as I guess we want to release it, can you please update the changelog so that it can be realeased ?

Done

franck-fourel commented on CHANGELOG.md:
```diff
@@ -1,6 +1,6 @@
-## Unreleased
+## 0.10.0 2020-09-09
 ### Changed 
 - PT-1435 latform services should be able to start without MongoDb
```
```suggestion
- PT-1435 Platform services should be able to start without MongoDb
```

franck-fourel committed:
Update CHANGELOG.md
